tests/suite/upgrade_test.go

```diff
@@ -229,15 +233,12 @@ var _ = Describe("Upgrade testing", Label("upgrade"), func() {
 			},
 		)).To(Succeed())
 
+		wg.Wait()
+		close(metricsCh)
+
 		recvdMetrics := make([]metricsResults, 0, 2)
```

Comment:
You don't really need this anymore. You can just range over the channel and write the results.